### PR TITLE
network: support matching based on wireless interface type

### PR DESCRIPTION
**Pull request by yuwata**
C.f. https://github.com/systemd/systemd/pull/13142#issuecomment-530540887.

Blocked by #13142.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
Azure Pipelines:

```
FAILED: clang -Isrc/libsystemd-network/efd04ff@@systemd-network@sta -Isrc/libsystemd-network -I../src/libsystemd-network -Isrc/basic -I../src/basic -Isrc/boot -I../src/boot -Isrc/shared -I../src/shared -Isrc/systemd -I../src/systemd -Isrc/journal -I../src/journal -Isrc/journal-remote -I../src/journal-remote -Isrc/nspawn -I../src/nspawn -Isrc/resolve -I../src/resolve -Isrc/timesync -I../src/timesync -I../src/time-wait-sync -Isrc/login -I../src/login -Isrc/udev -I../src/udev -Isrc/libudev -I../src/libudev -Isrc/core -I../src/core -Isrc/shutdown -I../src/shutdown -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -I. -I../ -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=gnu99 -g -fsanitize=fuzzer-no-link -Wextra -Werror=undef -Wmissing-include-dirs -Wold-style-definition -Wpointer-arith -Winit-self -Wfloat-equal -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=missing-declarations -Werror=return-type -Werror=incompatible-pointer-types -Werror=format=2 -Wstrict-prototypes -Wredundant-decls -Wmissing-noreturn -Wshadow -Wendif-labels -Wstrict-aliasing=2 -Wwrite-strings -Werror=overflow -Werror=shift-count-overflow -Wdate-time -Wnested-externs -Wno-unused-parameter -Wno-missing-field-initializers -Wno-unused-result '-Wno-error=#warnings' -Wno-string-plus-int -Wno-error=nonnull -ffast-math -fno-common -fdiagnostics-show-option -fno-strict-aliasing -fvisibility=hidden -fstack-protector -fstack-protector-strong --param=ssp-buffer-size=4 -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=shadow -include config.h -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fPIC -MD -MQ 'src/libsystemd-network/efd04ff@@systemd-network@sta/network-internal.c.o' -MF 'src/libsystemd-network/efd04ff@@systemd-network@sta/network-internal.c.o.d' -o 'src/libsystemd-network/efd04ff@@systemd-network@sta/network-internal.c.o' -c ../src/libsystemd-network/network-internal.c
../src/libsystemd-network/network-internal.c:151:10: error: array designator index (11) exceeds array bounds (11)
        [NL80211_IFTYPE_OCB] = "ocb",
         ^~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
##[error]Bash exited with code '1'.
```

**Comment by yuwata** [comment]
Currently, I use `WirelessType=`, but maybe, `WLANInterfaceType=` is suitable. See https://github.com/systemd/systemd/pull/13142#discussion_r327006141. WDYT?

**Comment by keszybz** [comment]
> Currently, I use `WirelessType=`, but maybe, `WLANInterfaceType=` is suitable. See [#13142 (comment)](https://github.com/systemd/systemd/pull/13142#discussion_r327006141). WDYT?

Yes, I think that is better. But not because of trademark reasons, but because "WirelessType" is so generic, and adding "Interface" there makes it more informative.

This looks pretty good. Can you do this one change and rebase?

**Comment by yuwata** [comment]
Rebased. PTAL.

File: man/systemd.network.xml [hunk 1]
```diff
@@ -153,6 +153,18 @@
             </para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>WirelessType=</varname></term>
```
Comment (keszybz) [inline man/systemd.network.xml:157]: The name needs to be changed here too.

Comment (yuwata) [reply man/systemd.network.xml:157]: Ouch. Thanks.

**Pull request merged**